packages/data-service/src/graphqlObservable.js

```diff
@@ -0,0 +1,92 @@
+import { Observable } from "rxjs/Observable";
+
+// TODO: refactor this once it has a more compreehensive implementation
+// of the graphql api
+
+// WARNING: This is NOT a spec complete graphql implementation
+// https://facebook.github.io/graphql/October2016/
+
+// eslint-disable-next-line import/prefer-default-export
+export const graphqlObservable = (doc, schema, context) => {
+  const translateOperation = {
+    query: "Query"
+  };
+
+  const resolveStep = (typeMap, definition, context, parent) => {
+    if (definition.kind === "OperationDefinition") {
+      const nextTypeMap = typeMap[
```

Comment:
This seems brittle, there are 4 null checks missing, do we refactor this later? TS could help here (at least me to not do dump comments like this ;) )

Reply:
`graphql-tools` ✅ 
`graphql-tag` :trollface:

Reply:
Could we move forward by saying that we will address that as we go:
https://jira.mesosphere.com/browse/DCOS-22006

To the point of trusting our implementation, I would also like to bring the tests form the graphql library into our implementation.